apps/api/scripts/delete-competitions.ts

```diff
@@ -0,0 +1,566 @@
+          )
+          .where(eq(competitionsLeaderboard.competitionId, competitionId))
+          .then((r) => r[0]?.count || 0);
+        deletions.push({
+          table: "trading_competitions_leaderboard",
+          count: Number(tradingLeaderboardDeleted),
+        });
+
+        // Delete from competitions_leaderboard
+        const leaderboardDeleted = await tx
+          .delete(competitionsLeaderboard)
+          .where(eq(competitionsLeaderboard.competitionId, competitionId))
+          .returning({ id: competitionsLeaderboard.id });
+        deletions.push({
+          table: "competitions_leaderboard",
+          count: leaderboardDeleted.length,
+        });
+      }
+
+      // Delete from competition_agents
+      const agentsDeleted = await tx
+        .delete(competitionAgents)
+        .where(eq(competitionAgents.competitionId, competitionId))
+        .returning({ agentId: competitionAgents.agentId });
+      deletions.push({
+        table: "competition_agents",
+        count: agentsDeleted.length,
+      });
+
+      // Delete from trading_competitions
+      const tradingCompDeleted = await tx
+        .delete(tradingCompetitions)
+        .where(eq(tradingCompetitions.competitionId, competitionId))
+        .returning({ id: tradingCompetitions.competitionId });
+      deletions.push({
+        table: "trading_competitions",
+        count: tradingCompDeleted.length,
+      });
+
+      // Finally, delete from competitions
+      const compDeleted = await tx
+        .delete(competitions)
+        .where(eq(competitions.id, competitionId))
+        .returning({ id: competitions.id });
+      deletions.push({ table: "competitions", count: compDeleted.length });
+
+      if (dryRun) {
+        // Rollback transaction in dry run mode
+        throw new Error("DRY_RUN_ROLLBACK");
+      }
+    });
+  } catch (error: unknown) {
+    if (error instanceof Error && error.message === "DRY_RUN_ROLLBACK") {
+      console.log(chalk.yellow("[DRY RUN] Transaction rolled back"));
+    } else {
+      throw error;
+    }
+  }
+
+  // Log deletion summary
+  console.log(
+    chalk.green(`\nDeletion summary for competition ${competitionId}:`),
+  );
+  deletions.forEach(({ table, count }) => {
+    if (count > 0) {
+      console.log(chalk.gray(`  - ${table}: ${count} rows`));
+    }
+  });
+
+  return deletions;
+}
+
+/**
+ * Recalculate agent scores after deleting ended competitions
+ */
+async function recalculateAgentScores() {
+  console.log(chalk.blue("\nRecalculating agent scores..."));
+
+  try {
+    // Get all remaining competitions that have ended
+    const endedCompetitions = await db
+      .select()
+      .from(competitions)
+      .where(eq(competitions.status, "ended"))
+      .orderBy(competitions.endDate);
+
+    console.log(
+      chalk.gray(
+        `Found ${endedCompetitions.length} ended competitions to process`,
+      ),
+    );
+
+    // Clear existing agent scores
+    await db.delete(agentScore);
```

Comment:
it is not deleting the agent_score_history, and at the end is also not adding entries to it